synopsys-task/test/unit/main.spec.ts

```diff
@@ -157,4 +158,33 @@ describe("Main function test cases", () => {
             })
         });
     });
+
+    context('return status', () => {
+        it('should extract the return_status from the error message', async () => {
+            const error = new Error('Requires at least one scan type '.concat(ErrorCode.MISSING_AT_LEAST_ONE_SCAN_TYPE.toString()));
+            const emptyErrorMessage = new Error('')
+
+            const returnStatus = main.getStatusFromError(error);
+            const emptyReturnStatus = main.getStatusFromError(emptyErrorMessage)
+
+            expect(returnStatus).to.equal(ErrorCode.MISSING_AT_LEAST_ONE_SCAN_TYPE.toString());
+            expect('').to.equal(emptyReturnStatus);
+        });
+    });
+
+    context('log exit codes', () => {
+        it('log corresponding error message including return_status', async () => {
+            const errorMessage = 'Requires at least one scan type'
+            const exitCode = ErrorCode.MISSING_AT_LEAST_ONE_SCAN_TYPE.toString();
+
+            const undefinedErrorMessage = 'Unknown error'
+            const undefinedExitCode = "9090";
+
+
+            const errorMessageForDefinedExitCodes = main.getExitMessage(errorMessage, exitCode);
+            const errorMessageForUndefinedExitCodes = main.getExitMessage(undefinedErrorMessage, undefinedExitCode);
+            expect("Exit Code: 101 - Requires at least one scan type").to.equal(errorMessageForDefinedExitCodes);
```

Comment:
why are we asserting 2 different error messages?

Reply:
Sometimes we may get undefined errors from extension for some scenario, for that case we are asserting the Undefined and Defined errors separately. But I think it can be achieved in 2 different unit tests.

Reply:
Separated these in different unit test cases